src/dswx_sar/pre_processing.py

```diff
@@ -124,111 +128,363 @@ def relocate(self,
-        reftif = gdal.Open(ref_file, gdal.GA_ReadOnly)
-        lat0, lon0 = self._read_x_y_array_geotiff(ref_file)
-        xsize = reftif.RasterXSize
-        ysize = reftif.RasterYSize
-        geotransform = reftif.GetGeoTransform()
-        proj = osr.SpatialReference(wkt=reftif.GetProjection())
-        epsg_output = proj.GetAttrValue('AUTHORITY',1)
-        xspacing = geotransform[1]
-        yspacing = geotransform[5]
-
-        reftif = None
-
-        if (len(lat0) != ysize) and (len(lon0) != xsize):
-
-            N, S, W, E = [np.max(lat0) + yspacing / 2,
-                          np.min(lat0) - yspacing / 2,
-                          np.min(lon0) - xspacing / 2,
-                          np.max(lon0) + xspacing / 2]
-
-            print('    Note: latitude shape is not same as image shape')
-
-        else:
-            N, S, W, E = [np.max(lat0),
-                          np.min(lat0),
-                          np.min(lon0),
-                          np.max(lon0)]
 
-        print('   bounding box', N, S , W, E)
-
-        # Crop (gdalwarp)image based on geo infomation of reference image
-        if yspacing < 0:
-            yspacing = -1 * yspacing
-
-        opt = gdal.WarpOptions(dstSRS=f'EPSG:{epsg_output}',
-                               xRes=xspacing,
-                               yRes=yspacing,
-                               outputBounds=[W, S, E, N],
-                               resampleAlg=method,
-                               format='GTIFF',
-                               dstNodata=no_data)
-
-        gdal.Warp(output_tif_str, input_tif_str, options=opt)
-
-
-    def _read_x_y_array_geotiff(self, intput_tif_str):
-        """Read X and Y coordinates from given Geotiff image
+        # forces returned values from TransformPoint() to be (x, y, z)
+        # rather than (y, x, z) for geographic SRS
+        if polygon_srs.IsGeographic():
+            try:
+                polygon_srs.SetAxisMappingStrategy(osr.OAMS_TRADITIONAL_GIS_ORDER)
+            except AttributeError:
+                logger.warning('WARNING Could not set the ancillary input SRS axis'
+                            ' mapping strategy (SetAxisMappingStrategy())'
+                            ' to osr.OAMS_TRADITIONAL_GIS_ORDER')
+        transformation = osr.CoordinateTransformation(tile_srs, polygon_srs)
+
+        elevation = 0
+        tile_x_array = np.zeros((4))
+        tile_y_array = np.zeros((4))
+        tile_x_array[0], tile_y_array[0], z = transformation.TransformPoint(
+            tile_min_x_utm, tile_max_y_utm, elevation)
+        tile_x_array[1], tile_y_array[1], z = transformation.TransformPoint(
+            tile_max_x_utm, tile_max_y_utm, elevation)
+        tile_x_array[2], tile_y_array[2], z = transformation.TransformPoint(
+            tile_max_x_utm, tile_min_y_utm, elevation)
+        tile_x_array[3], tile_y_array[3], z = transformation.TransformPoint(
+            tile_min_x_utm, tile_min_y_utm, elevation)
```

Comment:
Less boilerplate but just not verbose. @jungkyoJung your call if you want to adopt this.
```suggestion
        # init list of x and y tiles to match expected order to tile corners
        x_arrs = [tile_min_x_utm, tile_max_x_utm, tile_max_x_utm, tile_min_x_utm]
        y_arrs = [tile_max_y_utm, tile_max_y_utm, tile_min_y_utm, tile_min_y_utm]
        
        # transform corners
        for i, (x_arr, y_arr) in enumerate(zip(x_arrs, y_arrs)):        
            tile_x_array[i], tile_y_array[i], _ = transformation.TransformPoint(
                x_arr, y_arr, elevation)                
```